### PR TITLE
This Week Widget

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #NA

To test:
- build/run
- make sure you can install This Week Widgets of both medium and large size
- make sure the data in the widget look consistent with the app

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/WordPressStatsWidgets/Remote service/StatsWidgetsService.swift
Comment (frosty) [inline WordPress/WordPressStatsWidgets/Remote service/StatsWidgetsService.swift:161]: I'm not sure if we've changed this to `main` elsewhere now?

File: WordPress/Classes/Stores/StatsPeriodStore.swift
Comment (frosty) [inline WordPress/Classes/Stores/StatsPeriodStore.swift]: What's the purpose of reloading the timelines here and in the `.week` case below? Do we not just want to populate the data if we have new data and ignore it otherwise? Otherwise, based on my testing, we're calling reloadTimelines every time the user accesses any week page in the Stats section.

File: WordPress/WordPressStatsWidgets/Views/Cards/ListRow.swift
Comment (frosty) [inline WordPress/WordPressStatsWidgets/Views/Cards/ListRow.swift]: I think we might actually want this to be neutral for today or any day in the future – otherwise we get into this situation with sites with timezones ahead of our own:
IMG_0012

(In the app, the % and +/- for the 12th would be neutral)

File: WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift
Comment (frosty) [inline WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift]: Typo: weel -> week

File: WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift
Comment (frosty) [inline WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift]: Typo: peview -> preview

File: WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift
Comment (frosty) [inline WordPress/WordPressStatsWidgets/Views/Localization/LocalizableStrings.swift:72]: all time -> this week

**Review by frosty — COMMENTED** [review]
This is mostly working really well for me, and looks good.

Two other things to note during my testing:

- I ran into the 'you need to log in' issue, even though I was logged into the app (but I hadn't viewed any stats in the app yet)
- The first time I added the widget, I had a very poor (or not working) connection, so nothing had loaded properly. This is the widget I saw:

IMG_0011

I wonder if we need to add placeholder days in this situation, if we know the site but have no data. Or if we top align the text and add a 'Unable to load stats data' style message in the center of the widget (this may be the clearest option).

**Comment by Gio2018** [comment]
Thanks for reviewing @frosty ! I fixed most of the comments. A couple of them will be fixed once we merge the changes in [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15838) PR, so I think we should wait to merge this one until that one is merged

File: WordPress/Classes/Stores/StatsPeriodStore.swift
Comment (Gio2018) [reply WordPress/Classes/Stores/StatsPeriodStore.swift]: Agreed, we should review this once we merge [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15838)

File: WordPress/WordPressStatsWidgets/Remote service/StatsWidgetsService.swift
Comment (Gio2018) [reply WordPress/WordPressStatsWidgets/Remote service/StatsWidgetsService.swift:161]: same here, we should review this once we merge [this](https://github.com/wordpress-mobile/WordPress-iOS/pull/15838)

File: WordPress/Classes/Stores/StatsPeriodStore.swift
Comment (frosty) [inline WordPress/Classes/Stores/StatsPeriodStore.swift]: I wonder if it might be best to use something like `NSCalendar.current.date(byAdding: .day, value: -index, to: Date())` rather than hardcoding seconds per day?

**Review by frosty — COMMENTED** [review]
I agree, let's wait for the other PR – one quick comment on the placeholder dates!

File: WordPress/Classes/Stores/StatsPeriodStore.swift
Comment (Gio2018) [reply WordPress/Classes/Stores/StatsPeriodStore.swift]: Done

**Review by frosty — APPROVED** [review]
I've retested with a couple of accounts, and all looking good to me!

**Pull request merged**
